backend/backend-gen/src/main/java/eu/socialedge/hermes/backend/gen/PdfGenerator.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Hermes - The Municipal Transport Timetable System
+ * Copyright (c) 2016-2017 SocialEdge
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ */
+package eu.socialedge.hermes.backend.gen;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.Velocity;
+import org.apache.velocity.exception.ParseErrorException;
+import org.apache.velocity.exception.ResourceNotFoundException;
+import org.apache.velocity.runtime.RuntimeConstants;
+import org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader;
+
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+
+import eu.socialedge.hermes.backend.gen.data.StationScheduleTemplate;
+import lombok.val;
+
+public class PdfGenerator {
+    private static final String TEMPLATES_FOLDER = "templates";
```

Comment:
Don't you think it'll be better to extract interface here? For example:
```
interface ScheduleSerializer {
   byte[] serialize(Schedule schedule)
}
```
and use export type specific realizations (w/o factories):
```
class PdfScheduleSerializer implements ScheduleSerializer{
   // factory method where you pass template and other configurations / constructor
   byte[] serialize(Schedule schedule) {
      // impl
   }
}
```
or even simpler, configure `PdfScheduleSerializer` in `PdfGenerationConfig` coz template will be rather static than dynamically configurable

Reply:
Regarding `StationScheduleTemplate`, I think it'll be better to hide it and don't use in the `SchedulePdfGenerator` "service" because of being velocity specific.
I'm also wondering why can't we just use Schedule on it's own? If you need some additional params (`Line`, `Station`), just add overloaded `serialize` method to `ScheduleSerializer`.

Reply:
Because it will be extremely painful to parse inside template. I don't want such logic in template anyway